### PR TITLE
Update to add repo user-guide to CI automation

### PR DESCRIPTION
**Pull request by mazzystr**
Updates to add standard Prow jobs to user-guide repo CI

This is my first time working with Prow.  I copied what we had for kubevirt.github.io.  Hopefully I got it correct.

### CODE REVIEW

**Comment by mazzystr** [comment]
@codificat  I swear I took some time to look through project-infra for Prow stuff.  This look I found everything right away this time.

@dhiller ping

**Comment by mazzystr** [comment]
@codificat  It looks like we have some bad settings in the user-guide repo.  Prow check config is mad.

```
 {"component":"unset","file":"prow/cmd/checkconfig/main.go:78","func":"main.reportWarning","level":"warning","msg":"the following enable strict branchprotection context requirements even though Tide handles merges: [repo: kubevirt/user-guide]. See #5: https://github.com/kubernetes/test-infra/blob/master/prow/cmd/tide/maintainers.md#best-practices Also note that this validation is imperfect, see the check-config code for details","severity":"warning","time":"2021-01-12T22:57:27Z"}
{"component":"unset","file":"prow/cmd/checkconfig/main.go:81","func":"main.reportWarning","level":"fatal","msg":"Strict is set and there were warnings","severity":"fatal","time":"2021-01-12T22:57:27Z"} 
```

File: github/ci/prow/files/config.yaml [hunk 1]
```diff
@@ -98,6 +98,7 @@ tide:
     kubevirt/csi-driver: squash
     kubevirt/ssp-operator: squash
     kubevirt/must-gather: squash
+    kubevirt/user-guide: merge
```
Comment (dhiller) [inline github/ci/prow/files/config.yaml:101]: I figure this stands in conflict with the rules applied here: https://github.com/kubevirt/project-infra/blob/master/github/ci/prow/files/config.yaml#L292

See point 5 paragraph from [best practices](https://github.com/kubernetes/test-infra/blob/master/prow/cmd/tide/maintainers.md#best-practices) for how to handle the situation here.

File: github/ci/prow/files/jobs/kubevirt/user-guide/user-guide-periodics.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+periodics:
+  - interval: 24h
```
Comment (dhiller) [inline github/ci/prow/files/jobs/kubevirt/user-guide/user-guide-periodics.yaml:2]: What is the intent of this job? Is it general validation? If so, maybe rather use a postsubmit that runs everytime a push against `source` occurs.

File: github/ci/prow/files/jobs/kubevirt/user-guide/user-guide-periodics.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        base_ref: source
+        path_alias: user-guide
+    spec:
+      nodeSelector:
+        region: primary
+      containers:
+        - image: docker.io/library/python3
+          command: ["/bin/sh", "-c"]
+          args: ["echo &&
+                  mkdir -p /opt/buildhome/python3 &&
```
Comment (dhiller) [inline github/ci/prow/files/jobs/kubevirt/user-guide/user-guide-periodics.yaml:17]: Maybe create an image that has all the installs built in?

File: github/ci/prow/files/jobs/kubevirt/user-guide/user-guide-presubmits.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      always_run: true
+      skip_report: false
+      spec:
+        nodeSelector:
+          region: primary
+        containers:
+          - image: docker.io/library/python
+            command: ["/bin/sh", "-c"]
+            args: ["echo &&
+                    mkdir -p /opt/buildhome/python3 &&
```
Comment (dhiller) [inline github/ci/prow/files/jobs/kubevirt/user-guide/user-guide-presubmits.yaml:14]: The image from above could also be used here.

**Review by dhiller — COMMENTED** [review]
In general you need to tweak your branch protection settings, otherwise I don't really understand how the changes get into the repo. Please see my comments inline.

File: github/ci/prow/files/jobs/kubevirt/user-guide/user-guide-periodics.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+                  source /opt/buildhome/python3/bin/activate &&
+                  pip3 install mkdocs mkdocs-awesome-pages-plugin &&
+                  echo '*** BEGIN mkdocs.yml ***' &&
+                  cat /opt/build/repo/mkdocs.yml &&
+                  echo '*** END mkdocs.yml ***' &&
+                  mkdocs build -f mkdocs.yml -d site"]
```
Comment (dhiller) [inline github/ci/prow/files/jobs/kubevirt/user-guide/user-guide-periodics.yaml:23]: IIUC the mkdocs documentation mkdocs is not directly supported as Jekyll is for GH Pages, you need to invoke the step `mkdocs gh-deploy` for this to happen. Or does this happen elsewhere?

**Comment by mazzystr** [comment]
@dhiller Can you take a look at the `kubevirt.github.io` and `user-guide` settings?  @codificat and I (more Pep than I) reviewed the settings for both repos.  We think all the settings are correct and don't understand why check-prow-config fails.  

There is also something wrong with the permissions for both repos.  Pep sees the vast array of settings and I only see two setting categories.  We both should see the same thing.

**Comment by dhiller** [comment]
> @dhiller Can you take a look at the `kubevirt.github.io` and `user-guide` settings? @codificat and I (more Pep than I) reviewed the settings for both repos. We think all the settings are correct and don't understand why check-prow-config fails.
> 
> There is also something wrong with the permissions for both repos. Pep sees the vast array of settings and I only see two setting categories. We both should see the same thing.

**TL;DR: Remove ll.292-296 from https://github.com/kubevirt/project-infra/blob/master/github/ci/prow/files/config.yaml to get rid of the settings for GitHub**

Problem here is that the [settings for branch-protector here](https://github.com/kubevirt/project-infra/blob/master/github/ci/prow/files/config.yaml#L292) contains [instructions for GitHub](https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#update-branch-protection), where it tells GitHub to check whether the branch is mergeable.

This conflicts with [Tide](https://github.com/kubernetes/test-infra/tree/master/prow/tide#tide), who will try to automatically rebase, test and merge your PR should `lgtm` and `approve` labels be present. But it won't be able to, because GitHub will interfere with this.

I hope that explanation makes sense, please ping me if it's unclear.

**Review by dhiller — APPROVED** [review]
/lgtm
/approve

**Comment by dhiller** [comment]
/override prow-deploy-test

Unrelated.

**Comment by mazzystr** [comment]
Thank you!
